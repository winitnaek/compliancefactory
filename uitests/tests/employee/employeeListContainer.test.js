import React from 'react';

import EmployeeList from '../../../src/app/employee/EmployeeListContainer';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

 const mockStore = configureMockStore();
 
 describe('Employee rows renders correctly', () => {
  let store
  beforeEach(function() {
    store = mockStore();
  }); 

  it('renders correctly', () => {
      store = mockStore({emps: [
      { "first_name": "Test First Name",
      "last_name": "Test Last Name",
      "email": "testemail@email.com",
      "id": 1}]});
    const rendered = renderer.create(
      <Provider store={store}>
      <EmployeeList/>
      </Provider>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('renders Loading to test empty employees', () => {
    const rendered = renderer.create(
      <Provider store={store}>
      <EmployeeList/>
      </Provider>
    );
    expect(rendered.toJSON()).toMatchSnapshot(<div>...Loaidng</div>);
  });
});