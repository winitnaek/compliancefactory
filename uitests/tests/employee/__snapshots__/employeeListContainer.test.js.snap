// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
<div>
  ...Loaidng
</div>
`;

exports[`Employee rows renders correctly renders correctly 1`] = `
<div
  className="cBox boxTools col-md-8 col-md-offset-2 col-sm-10 col-sm-offset-1"
>
  <h1>
    Employee List
  </h1>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Id
        </th>
        <th>
          First Name
        </th>
        <th>
          Last Name
        </th>
        <th>
          Email
        </th>
        <th>
          Delete
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          Test First Name
        </td>
        <td>
          Test Last Name
        </td>
        <td>
          testemail@email.com
        </td>
        <td>
          <button
            className="btn btn-default"
            onClick={[Function]}
          >
            Delete me!
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
